openedx/adg/common/mailchimp_pipeline/tests/test_helpers.py

```diff
@@ -0,0 +1,119 @@
+"""
+All tests for mailchimp pipeline helpers
+"""
+from datetime import datetime
+
+import pytest
+
+from openedx.adg.common.course_meta.tests.factories import CourseMetaFactory
+from openedx.adg.common.mailchimp_pipeline import helpers as mailchimp_helpers
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+
+
+@pytest.fixture
+def user_enrollments():
+    """
+    A fixture for enrolling user into multiple running courses.
+    """
+    user = UserFactory()
+    enrollment_data = {'user': user, 'is_active': True, 'course__end_date': datetime(2022, 1, 1)}
+    enrollment1 = CourseEnrollmentFactory(course__display_name='course1', **enrollment_data)
+    enrollment2 = CourseEnrollmentFactory(course__display_name='course2', **enrollment_data)
+    enrollment3 = CourseEnrollmentFactory(course__display_name='course3', **enrollment_data)
+    enrolled_courses = [enrollment1.course, enrollment2.course, enrollment3.course]
+
+    for course in enrolled_courses:
+        CourseMetaFactory(course=course)
+
+    return user, enrolled_courses
+
+
+@pytest.mark.django_db
+def test_get_active_enrollment_course_overviews_by_user(user_enrollments):  # pylint: disable=redefined-outer-name
+    """
+    Assert that all active and running enrolled courses for a user are returned.
+    """
+    user, enrolled_courses = user_enrollments
+    expected_courses = mailchimp_helpers.get_active_enrollment_course_overviews_by_user(user)
+
+    assert expected_courses == enrolled_courses
+
+
+@pytest.mark.django_db
+def test_get_active_enrollment_course_names_by_user(mocker, user_enrollments):  # pylint: disable=redefined-outer-name
+    """
+    Assert that all active and running enrolled courses for a user are returned.
+    """
+    user, enrolled_courses = user_enrollments
+    mock_courses = mocker.patch.object(mailchimp_helpers, 'get_active_enrollment_course_overviews_by_user')
+    mock_courses.return_value = enrolled_courses
+    expected_course_titles = mailchimp_helpers.get_active_enrollment_course_names_by_user(user)
+
+    assert expected_course_titles == 'course1, course2, course3'
+
+
+@pytest.mark.django_db
+def test_get_active_enrollment_course_short_ids(mocker, user_enrollments):  # pylint: disable=redefined-outer-name
+    """
+    Assert that all active and running enrolled courses for a user are returned.
+    """
+    user, enrolled_courses = user_enrollments
+    mock_courses = mocker.patch.object(mailchimp_helpers, 'get_active_enrollment_course_overviews_by_user')
+    mock_courses.return_value = enrolled_courses
+    expected_course_short_ids = mailchimp_helpers.get_active_enrollment_course_short_ids(user)
```

Comment:
Same expected is confusing and I guess we don't need to mock here.

Reply:
done